src/compiler/transformers/classFields.ts

```diff
@@ -137,14 +179,217 @@ namespace ts {
+                        createBinary(
+                            node.operand,
+                            operator,
+                            createNumericLiteral("1")
+                        ),
+                        node
+                    );
+                    const visited = visitNode(transformedExpr, visitor);
+                    // If the private name was successfully transformed,
+                    // return the transformed node. Otherwise, leave existing source untouched.
+                    if (visited !== transformedExpr) {
+                        return visited;
+                    }
+                }
+            }
+            return visitEachChild(node, visitor, context);
+        }
+
+        function visitPostfixUnaryExpression(node: PostfixUnaryExpression) {
+            if (isPrivateIdentifierPropertyAccessExpression(node.operand)) {
+                const operator = (node.operator === SyntaxKind.PlusPlusToken) ?
+                    SyntaxKind.PlusToken : (node.operator === SyntaxKind.MinusMinusToken) ?
+                        SyntaxKind.MinusToken : undefined;
+                if (operator) {
+                    // Create a temporary variable if the receiver is not inlinable, since we
+                    // will need to access it multiple times.
+                    const receiver = isSimpleInlineableExpression(node.operand.expression) ?
+                        undefined :
+                        getGeneratedNameForNode(node.operand.expression);
+                    // Create a temporary variable to store the value returned by the expression.
+                    const returnValue = createTempVariable(/*recordTempVariable*/ undefined);
+
+                    const transformedExpr = createCommaList(compact([
+                        receiver && createAssignment(receiver, node.operand.expression),
+                        // Store the existing value of the private name in the temporary.
+                        createAssignment(returnValue, receiver ? createPropertyAccess(receiver, node.operand.name) : node.operand),
+                        // Assign to private name.
+                        createAssignment(
+                            receiver ? createPropertyAccess(receiver, node.operand.name) : node.operand,
+                            createBinary(
+                                returnValue, operator, createNumericLiteral("1")
+                            )
+                        ),
+                        // Return the cached value.
+                        returnValue
+                    ]) as Expression[]);
+                    const visited = visitNode(transformedExpr, visitor);
+                    // If the private name was successfully transformed,
+                    // hoist the temporary variable and return the transformed node.
+                    // Otherwise, leave existing source untouched.
+                    if (visited !== transformedExpr) {

```

Comment:
It now always results in a change (since we replace private names with missing identifiers for private names which are not declared in the lexical scope), so I removed this check.